### ISSUE TITLE
[Suggestion] support imported `.yaml` with mixed CRD and normal resource definitions

### ISSUE THREAD

**Issue by campionfellin** [awslabs/cdk8s#109]
**Is your feature request related to a problem? Please describe.**

When importing resources, we generally expect them to be either CRDs or `k8s`. We should support a file that has both CRDs and standard resource definitions in them.

For example, take https://operatorhub.io/operator/composable-operator (can also be found here: https://operatorhub.io/api/operator?packageName=composable-operator)

This contains `Service` (from `k8s`), a CRD named `Composable`, and a `KafkaSource`  (from api version `sources.eventing.knative.dev/v1alpha1`).

**Describe the solution you'd like**

They should be able to import this file, and have 3 `imports/`:

`service.ts` (they shouldn't need all of `k8s` imported)
`composable.ts`  (on of the CRDs)
`kafkasource.ts` (a standard resource from another api version)

**Describe alternatives you've considered**

**Additional context**

This should unblock some work on https://github.com/awslabs/cdk8s/issues/72

This issue might be similar to https://github.com/awslabs/cdk8s/issues/71 but I think is distinct enough to have its own issue.

**Comment by eladb** [comment]
Interesting idea. What do you expect `service.ts` to include?

**Comment by campionfellin** [comment]
I'd expect it to have the minimum subset of `k8s` that you would need for it to work. So maybe:

```

export interface ServiceOptions {
  readonly metadata?: ObjectMeta;
  readonly spec?: ServiceSpec;
}

export class Service extends ApiObject {
  public constructor(scope: Construct, name: string, options: ServiceOptions = {}) {
    super(scope, name, {
      ...options,
      kind: 'Service',
      apiVersion: 'v1',
    });
  }
}

export interface ServiceSpec {
..
.
.
```

etc.
---
That being said, I am now thinking we may want to simplify and just detect and auto-import the whole `k8s` API for them (or in the case of the example, the `serving.knative.dev/v1` API).

**Comment by eladb** [comment]
@campionfellin I still don't fully understand this workflow. The referenced YAML does not declare the "Service" type, it just uses a resource of that type. `cdk8s import` is designed to generate types...

**Comment by campionfellin** [comment]
Re-reading this, it seems I was confused about the functionality of `cdk8s import` at the time. Closing.
